cpp/ql/src/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -324,3 +324,15 @@ class IteratorArrayMemberOperator extends MemberFunction, TaintFunction {
     output.isReturnValue()
   }
 }
+
+class BeginOrEndFunction extends MemberFunction, TaintFunction {
+  BeginOrEndFunction() {
+    this.hasName(["begin", "cbegin", "rbegin", "crbegin", "end", "cend", "rend", "crend"]) and
```

Comment:
This is great. Have we also modelled [`std::begin` and `std::cbegin`](https://en.cppreference.com/w/cpp/iterator/begin)?

Reply:
This model can replace `StdStringBeginEnd` and most of `StdSequenceContainerBeginEnd` (in https://github.com/github/codeql/pull/4201/files#diff-cbbb968ec9678cb6b97fb7001e0f7261R173).

Reply:
Ah, I see you've removed `StdStringBeginEnd` later in the PR.  I will take responsibility for removing `StdSequenceContainerBeginEnd` as well once both PRs have been merged.

Reply:
`StdSequenceContainerBeginEnd ` removed in https://github.com/github/codeql/pull/4395.